azure-keyvault/src/main/java/com/microsoft/azure/keyvault/messagesecurity/HttpMessageSecurity.java

```diff
@@ -0,0 +1,330 @@
+
+            byte[] aesKeyBytes = new byte[32];
+            new Random().nextBytes(aesKeyBytes);
+            if (testMode){
+                aesKeyBytes = "TEST1234TEST1234TEST1234TEST1234".getBytes();
+            }
+
+            SymmetricKey aesKey = new SymmetricKey(UUID.randomUUID().toString(), aesKeyBytes);
+
+            byte[] iv = new byte[16];
+            new Random().nextBytes(iv);
+            if (testMode){
+                iv = "TEST1234TEST1234".getBytes();
+            }
+
+            RsaKey serverEncryptionRsaKey = new RsaKey(serverEncryptionKey.kid(), serverEncryptionKey.toRSA(false));
+            Triple<byte[], byte[], String> encrypted_key = serverEncryptionRsaKey.encryptAsync(aesKeyBytes, null, null, "RSA-OAEP").get();
+
+            Triple<byte[], byte[], String> cipher = aesKey.encryptAsync(
+                    payload.getBytes(),
+                    iv,
+                    MessageSecurityHelper.stringToBase64Url(jweHeader.serialize()).getBytes(),
+                    "A128CBC-HS256").get();
+
+            JWEObject jweObject = new JWEObject(jweHeader,
+                    MessageSecurityHelper.bytesToBase64Url((!testMode) ? encrypted_key.getLeft() : "key".getBytes()),
+                    MessageSecurityHelper.bytesToBase64Url(iv),
+                    MessageSecurityHelper.bytesToBase64Url(cipher.getLeft()),
+                    MessageSecurityHelper.bytesToBase64Url(cipher.getMiddle()));
+
+            return jweObject;
+        } catch (ExecutionException e){
+            // unexpected;
+            return null;
+        } catch (InterruptedException e){
+            // unexpected;
+            return null;
+        } catch (NoSuchAlgorithmException e){
+            // unexpected;
+            return null;
+        }
+    }
+
+    /**
+     * Unencrypt encrypted payload.
+     *
+     * @param payload
+     *      base64url serialized JWEObject.
+     *
+     * @return
+     *      Unencrypted message.
+     */
+    private String unprotectPayload(String payload) throws IOException {
+        try {
+            JWEObject jweObject = JWEObject.deserialize(MessageSecurityHelper.base64UrltoString(payload));
+            JWEHeader jweHeader = jweObject.jweHeader();
+
+            if (!clientEncryptionKey.kid().equals(jweHeader.kid()) ||
+                    !jweHeader.alg().equals("RSA-OAEP") ||
+                    !jweHeader.enc().equals("A128CBC-HS256")) {
+                throw new IOException("Invalid protected response");
+            }
+
+            byte[] key = MessageSecurityHelper.base64UrltoByteArray(jweObject.encryptedKey());
+
+            RsaKey clientEncryptionRsaKey = new RsaKey(clientEncryptionKey.kid(), clientEncryptionKey.toRSA(true));
+            byte[] aesKeyBytes = (!testMode) ? clientEncryptionRsaKey.decryptAsync(key, null, null, null, "RSA-OAEP").get() : "TEST1234TEST1234TEST1234TEST1234".getBytes();
```

Comment:
>byte[] aesKeyBytes = (!testMode) ? clientEncryptionRsaKey.decryptAsync(key, null, null, null, "RSA-OAEP").get() : "TEST1234TEST1234TEST1234TEST1234".getBytes(); [](start = 12, length = 160)

shouldn't the decryption work even in testMode? i.e. if you decrypt jweObject.encryptedKey() in testMode shouldn't you get back "TEST1234...